test/session_tests/DocumentDeleteTest.ts

```diff
@@ -0,0 +1,66 @@
+/// <reference path="../../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../../node_modules/@types/chai/index.d.ts" />
+
+import {expect} from 'chai';
+import * as Promise from 'bluebird';
+import {IHash} from "../../src/Utility/Hash";
+import {DocumentStore} from "../../src/Documents/DocumentStore";
+import {IDocumentStore} from "../../src/Documents/IDocumentStore";
+import {string, Object} from "../../src/Documents/Object";
+import {IDocumentSession} from "../../src/Documents/Session/IDocumentSession";
+
+describe('Document delete test', () => {
```

Comment:
Please also include some tests using async/await syntax for basic CRUD, to verify that it works and give people some sample code to look at.